### PR TITLE
Correctly parse fixnums between -129 and -256.

### PR DESCRIPTION
**Pull request by philipcunningham**
Fixes https://github.com/instore/node-marshal/issues/2.

### CODE REVIEW

**Comment by philipcunningham** [comment]
Let me know if you think there's a better way of doing this!

File: lib/marshal.js [hunk 1]
```diff
@@ -83,7 +83,11 @@ Marshal = (function () {
     }
     else if (small === -1) {
       var large = this.buffer.readInt8(this._index++);
-      return large;
+      if (large >= 0 && large <= 127) {
+        return large - 256;
```
Comment (philipcunningham) [inline lib/marshal.js:87]: @clayzermk1 even though subtracting 256 from an Int8 will result in an overflow, JavaScript doesn't have an Int8 so it doesn't cause any problems.

**Comment by clayzermk1** [comment]
I think I'd prefer to do all negative integers by using a buffer mask like I did for negative three byte integers.

If it's all the same to you, do you mind if I author my commit in your name?

**Comment by philipcunningham** [comment]
> I think I'd prefer to do all negative integers by using a buffer mask like I did for negative three byte integers.

I can see how that would work. What do you think about the cost of creating a new buffer for every negative integer? 

> If it's all the same to you, do you mind if I author my commit in your name?

I don't mind as long as the commit message explains you've authored it under my name.

**Comment by clayzermk1** [comment]
Yeah, that cost is high compared to simple math. I want a general case though that I can use for all the negative numbers. Binary in Javascript is a pain, I'm going to look into doing it without the buffer creation. I think I can get away with it using `parseInt()`.

I'll be sure to blatantly say it. Thanks :smile:

**Comment by philipcunningham** [comment]
Sure, I can see it from that perspective too. 

Oh, one last thing. I just wanted to say thanks for writing this library and appreciate you taking the time to reply to me on this PR.

**Comment by clayzermk1** [comment]
I'm humbled that someone else was able to find it useful and I really appreciate you taking the time to report it and work with me :beers:

**Pull request closed without merging**